### PR TITLE
doc: Update doc versions for 2.5 release

### PR DESCRIPTION
**Pull request by umapraseeda**
Update doc versions for 2.5 release

### CODE REVIEW

File: doc/tfm/conf.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 project = "Trusted Firmware-M"
 copyright = "2017-2021, ARM CE-OSS"
 author = "ARM CE-OSS"
-version = "1.7.0"
+version = "1.8.0"
```
Comment (umapraseeda) [inline doc/tfm/conf.py:28]: @shanthanordic/ @carlescufi / @gmarull  please confirm if we need the version updates for TF-M MCUboot etc.

Comment (shanthanordic) [reply doc/tfm/conf.py:28]: Yes TFM is 1.8.0 Version

**Review by shanthanordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: doc/nrf/protocols/matter/index.rst [hunk 2]
```diff
@@ -20,9 +20,7 @@ The |NCS| allows you to develop applications with different versions of Matter,
 +--------------------------+-----------------------------------------------------+------------------------+
 | nRF Connect SDK version  | Matter specification version                        | `Matter SDK version`_  |
 +==========================+=====================================================+========================+
-| v2.4.99 (latest)         | :ref:`1.1.0 <ug_matter_overview_dev_model_support>` | 1.1.0.1                |
-+--------------------------+                                                     |                        |
-| |release|                |                                                     |                        |
+| |release|                | :ref:`1.1.0 <ug_matter_overview_dev_model_support>` | 1.1.0.1                |
```
Comment (divipillai) [inline doc/nrf/protocols/matter/index.rst:23]: Should v2.4.0 also be added here? I could see minor releases are also added in the table.

Comment (peknis) [reply doc/nrf/protocols/matter/index.rst:23]: This is a very good question. I'd believe we need to have 2.4.0 included. @greg-fer, what do you think?

Comment (peknis) [reply doc/nrf/protocols/matter/index.rst:23]: Updated the table to include 2.4.0 + patch releases.

Comment (wiba-nordic) [reply doc/nrf/protocols/matter/index.rst:23]: Discussed offline with @LuDuda and agreed all previous versions should be in the table. Worked with @peknis to add the values.

**Review by wiba-nordic — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

File: doc/nrf/protocols/matter/index.rst (hunk 2 above)
Comment (b-gent) [reply doc/nrf/protocols/matter/index.rst:23]: thanks

**Review by b-gent — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Pull request merged**